pkg/core/transformer/main_test.go

```diff
@@ -186,6 +186,32 @@ var (
 			expectedOutput: "1.17",
 			expectedErr:    nil,
 		},
+		Data{
+			input: "1.17.0",
+			rules: Transformers{
+				Transformer{
+					"findSubMatch": FindSubMatch{
+						Pattern:      `\d*.(\d*)`,
+						CaptureIndex: 1,
+					},
+				},
+			},
+			expectedOutput: "17",
+			expectedErr:    nil,
+		},
+		Data{
+			input: "1.17.0",
+			rules: Transformers{
+				Transformer{
+					"findSubMatch": FindSubMatch{
+						Pattern:      `\d*.\d*`,
+						CaptureIndex: 1,
+					},
+				},
+			},
+			expectedOutput: "",
+			expectedErr:    nil,
+		},
```

Comment:
I would also add a test with a too big CaptureIndex, and maybe another one with multiple submatches/capture groups (ex: `\d*.(\d*).(\d*)` (not sure if there is a use case for this one)

Reply:
the choice is to avoid any error (as per discussion with damien about this behavior when populating the issue), so I added a test but with no error match .... (it's a logrus log)